packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -70,12 +132,20 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 
 	public constructor(client: ApplicationInsights, config?: FluidAppInsightsLoggerConfig) {
 		this.baseLoggingClient = client;
+
 		this.config = config ?? {
 			filtering: {
 				mode: "exclusive",
 				filters: [],
 			},
 		};
+
+		if (this.config.filtering.filters) {
+			// Sorts filters in order from most specific to least specific.
+			this.config.filtering.filters = FluidAppInsightsLogger.sortTelemetryFilters(
+				this.config.filtering.filters,
+			);
+		}
```

Comment:
Now with the sorting, we probably want to save the sorted copy of the list in a local variable instead of updating the property; the `config` being an object passed by the caller, we probably don't want to mutate it in any way.

In fact... we might want to make a deep copy of it to store locally (and then it's fine to update the filter list in place), so that any changes that the caller might make to the object they pass in don't affect the logger.

Reply:
Agreed. I've updated it to create an internal deep copy filter array.